### PR TITLE
Build bf16 gemv fast path & entry points for non-ARM architectures too

### PR DESCRIPTION
**Pull request by swolchok**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #139220
* __->__ #139208
* #139081
* #139558
* #139090
* #139084

Very similar to #137917, but for bf16.

Differential Revision: [D65155971](https://our.internmc.facebook.com/intern/diff/D65155971/)

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Comment by swolchok** [comment]
will look into failures here after I fix #139090.

**Comment by swolchok** [comment]
suspect failures might go away now that we do the scalar tail fixup in float; we'll see next time I export

**Comment by swolchok** [comment]
looking into continued test failures

File: aten/src/ATen/native/cpu/ReducedPrecisionFloatGemvFastPathKernel.cpp [hunk 1]
```diff
@@ -156,17 +156,16 @@ float reduce(vec::VectorizedN<float, kF32RegistersPerIteration>& x) {
 // We would have to write a separate SVE-specific path to use SVE
 // BFDOT. Deferring that for now to get the NEON/ASIMD BFDOT path
 // working.
-#if defined(__aarch64__) && !defined(CPU_CAPABILITY_SVE)
-#if defined(__clang__) && __clang_major__ > 15
+#if defined(__aarch64__) && !defined(CPU_CAPABILITY_SVE) && defined(__clang__) && __clang_major__ > 15
 // https://godbolt.org/z/z8P4Yncra
 #define COMPILER_SUPPORTS_BF16_TARGET 1
-#elif !defined(__clang__) && defined(__GNUC__) && __GNUC__ >= 10
+#elif defined(__aarch64__) && !defined(CPU_CAPABILITY_SVE) && !defined(__clang__) && defined(__GNUC__) && __GNUC__ >= 10
```
Comment (malfet) [inline aten/src/ATen/native/cpu/ReducedPrecisionFloatGemvFastPathKernel.cpp:162]: Can this be moved to say `compiler_capabilites` header which is included form here, that has a table on top that explains which compiler versions supports what

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**